### PR TITLE
Seafile: Fix download/upload error when FILE_SERVER_ROOT is relative

### PR DESCRIPTION
**Pull request by DanielEgbers**
#### What is the purpose of this change?

This PR fixes an error in the Seafile backend implementation.
A Seafile server can be configured to use a relative url as `FILE_SERVER_ROOT` (e.g.: `/seafhttp` instead `https://seafile.example.com/seafhttp`) in order to support more than one hostname/ip (see https://github.com/haiwen/seahub/issues/3398#issuecomment-506920360).
The current backend implementation always expected an absolute download/upload url, resulting in an `unsupported protocol scheme` error. With this PR it supports both absolute and relative.

#### Was the change discussed in an issue or in the forum before?

No

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/seafile/webapi.go [hunk 1]
```diff
@@ -606,9 +606,14 @@ func (f *Fs) download(ctx context.Context, url string, size int64, options ...fs
 	// Build the http request
 	opts := rest.Opts{
 		Method:  "GET",
-		RootURL: url,
 		Options: options,
 	}
+	parsedURL, _ := url.Parse(downloadLink)
```
Comment (ncw) [inline backend/seafile/webapi.go:611]: This needs to check the error returned by `url.Parse` if err !=nil then `parsedURL` won't be valid.

File: backend/seafile/webapi.go [hunk 2]
```diff
@@ -688,11 +693,16 @@ func (f *Fs) upload(ctx context.Context, in io.Reader, uploadLink, filePath stri
 
 	opts := rest.Opts{
 		Method:      "POST",
-		RootURL:     uploadLink,
 		Body:        formReader,
 		ContentType: contentType,
 		Parameters:  url.Values{"ret-json": {"1"}}, // It needs to be on the url, not in the body parameters
 	}
+	parsedURL, _ := url.Parse(uploadLink)
```
Comment (ncw) [inline backend/seafile/webapi.go:700]: same here...

**Review by ncw — COMMENTED** [review]
This looks like a nice fix - thank you.

I think the errors from `url.Parse` need checking though...

File: backend/seafile/webapi.go (hunk 1 above)
Comment (DanielEgbers) [reply backend/seafile/webapi.go]: added error check

File: backend/seafile/webapi.go (hunk 2 above)
Comment (DanielEgbers) [reply backend/seafile/webapi.go]: added error check

**Review by ncw — APPROVED** [review]
That looks great now - thank you.

Will merge :-)

**Pull request merged**